deeppavlov/core/data/utils.py

```diff
@@ -437,21 +588,30 @@ def jsonify_data(data):
     return result
 
 
-def chunk_generator(items_list, chunk_size):
+def chunk_generator(items_list: list, chunk_size: int) -> Generator[list, None, None]:
+    """Yields consecutive slices of list.
+
+    Args:
+        items_list: List to slice.
+        chunk_size: Length of slice.
+
+    Yields:
+        list: `items_list` consecutive slices.
+
+    """
     for i in range(0, len(items_list), chunk_size):
         yield items_list[i:i + chunk_size]
 
 
-def update_dict_recursive(editable_dict: dict, editing_dict: dict) -> None:
-    """Updates dict recursively
+def update_dict_recursive(editable_dict: dict, editing_dict: Union[dict, Mapping]) -> None:
```

Comment:
```suggestion
def update_dict_recursive(editable_dict: dict, editing_dict: Mapping) -> None:
```
`dict` is a `Mapping`